app/src/cnm/java/org/eyeseetea/malariacare/domain/usecase/LoginUseCase.java

```diff
@@ -24,7 +24,7 @@ public void execute(final Credentials credentials, final Callback loginCallback)
                 new IAuthenticationManager.Callback<UserAccount>() {
                     @Override
                     public void onSuccess(UserAccount userAccount) {
-                        if (!credentials.isDemoCredentials()) {
+                        if (credentials == null && !credentials.isDemoCredentials()) {
```

Comment:
if credentials is null on execute we should throw IllegalArgumentException, so validate null onSuccess is not necesary